src/sql/parse.py

```diff
@@ -218,8 +219,18 @@ def without_sql_comment(parser, line):
 
 
 def magic_args(magic_execute, line):
+    """Function to parse the magic arguments. Returns None if
+    mandatory positional argument not provided"""
     line = without_sql_comment(parser=magic_execute.parser, line=line)
-    return magic_execute.parser.parse_args(shlex.split(line, posix=False))
+    split = shlex.split(line, posix=False)
+    parser = magic_execute.parser
+    try:
+        return parser.parse_args(split)
+    except UsageError as e:
+        if f"the following arguments are required: {parser._actions[0].dest}" in str(e):
+            return None
+        else:
+            raise
```

Comment:
this stills too hacky (the only reason we are allowing this pattern in other parts of the codebase is because we have no control over the database driver errors), it's better to rely on the argparser logic. it's fine even if the error doesn't show the available choices when the argument is missing

Reply:
sure will remove it . I think we don't need to add `choices` since there's already some custom logics for plot name validation. wdyt? @edublancas 

https://github.com/ploomber/jupysql/blob/1a090c16e8029861324bf0f17027582152343db8/src/sql/magic_plot.py#L98

Reply:
let's go with choices. better to leave validation to argparse than do it ourselves.

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/839/commits/18baa65c025c7b26b5d515f5c0cb450bbcb18b7d)